commit 98d2722a85c4ad5f2baf2272cbb0fab67f797b69
Author: Daniele Ceraolo Spurio <daniele.ceraolospurio@intel.com>
Date:   Wed May 31 16:54:12 2023 -0700

    drm/i915/huc: differentiate the 2 steps of the MTL HuC auth flow
    
    Before we add the second step of the MTL HuC auth (via GSC), we need to
    have the ability to differentiate between them. To do so, the huc
    authentication check is duplicated for GuC and GSC auth, with
    GSC-enabled binaries being considered fully authenticated only after
    the GSC auth step.
    
    To report the difference between the 2 auth steps, a new case is added
    to the HuC getparam. This way, the clear media driver can start
    submitting before full auth, as partial auth is enough for those
    workloads.
    
    v2: fix authentication status check for DG2
    
    v3: add a better comment at the top of the HuC file to explain the
        different approaches to load and auth (John)
    
    v4: update call to intel_huc_is_authenticated in the pxp code to check
    for GSC authentication
    
    v5: drop references to meu and esclamation mark in huc_auth print (John)
    
    Signed-off-by: Daniele Ceraolo Spurio <daniele.ceraolospurio@intel.com>
    Cc: Alan Previn <alan.previn.teres.alexis@intel.com>
    Cc: John Harrison <John.C.Harrison@Intel.com>
    Reviewed-by: Alan Previn <alan.previn.teres.alexis@intel.com> #v2
    Reviewed-by: John Harrison <John.C.Harrison@Intel.com>
    Link: https://patchwork.freedesktop.org/patch/msgid/20230531235415.1467475-5-daniele.ceraolospurio@intel.com

diff --git a/drivers/gpu/drm/i915/i915_reg.h b/drivers/gpu/drm/i915/i915_reg.h
index c4197e31962e..e0fc0f588c45 100644
--- a/drivers/gpu/drm/i915/i915_reg.h
+++ b/drivers/gpu/drm/i915/i915_reg.h
@@ -941,6 +941,9 @@
 #define HECI_H_GS1(base)	_MMIO((base) + 0xc4c)
 #define   HECI_H_GS1_ER_PREP	REG_BIT(0)
 
+#define HECI_FWSTS5(base)		_MMIO((base) + 0xc68)
+#define   HECI_FWSTS5_HUC_AUTH_DONE	(1 << 19)
+
 #define HSW_GTT_CACHE_EN	_MMIO(0x4024)
 #define   GTT_CACHE_EN_ALL	0xF0007FFF
 #define GEN7_WR_WATERMARK	_MMIO(0x4028)